developers/release.md

```diff
@@ -0,0 +1,233 @@
+
+### Step 0: Code Name
+
+Pick a release code name based on a style of beer previously unused. Examples from previous releases are Oktoberfest and Winter Warmer.
+
+### Step 1: Warn Community of Branching
+
+The first step to any release is warning the community of impending branching prior to the branch date to give them time to push on any pull requests that are pending. The email should be sent no less than 2 weeks prior to the intended branch date and ideally when Foreman sends their branching announcement.
+
+### Step 2: Stable Pulp Build
+
+Prior to branching the Katello repositories, the release nanny should figure out what the current stable version of Pulp is and if not already tagged and built, do so. Please see the [Upgrading Pulp Guide](http://www.katello.org/developers/upgrade_pulp.html).
+
+### Step 3: Setup Your Environment
+
+There are a number of repositories that will need to be tagged and branched throughout the process. We recommend organizing this before hand so that you have a common place to reference and work through these tasks within throughout the release process. Clone the following:
+
+```
+git clone https://github.com/Katello/katello
+git clone https://github.com/Katello/katello-installer.git
+git clone https://github.com/Katello/katello-misc
+git clone https://github.com/Katello/katello-agent
+git clone https://github.com/Katello/hammer-cli-katello
+```
+
+Note that some repositories contain multiple spec-files and thus packages:
+
+```
+katello
+  rubygem-katello.spec
+  deploy/katello.spec
+
+katello-misc
+  certs-tools/katello-certs-tools.spec
+  repos/katello-repos.spec
+  utils/katello-utils.spec
+
+katello-agent
+  katello-agent.spec
+
+katello-installer
+  katello-installer.spec
+
+hammer-cli-katello
+  rubygem-hammer_cli_katello.spec
+```
+
+### Step 4: Get Your Branch On
+
+On the date and time you previously warned the community branching would occur on, do the following for each repository EXCEPT `hammer-cli-katelo`.
+
+Create the X.Y branch
+
+```
+cd katello
+git fetch origin
+git checkout master
+git reset origin/master --hard
+tito tag --keep-version
+git branch -b KATELLO-X.Y
+```
+
+Update the releasers and tito configuration for each package (some repositories contain multiple packages, see Step 3 for list):
+
+```
+vim rel-eng/releasers.conf
+:tabnew rel-eng/tito.props
+
+:tabdo %s/katello-nightly-rhel6/katello-X.Y-rhel6/g
+:tabdo %s/katello-nightly-rhel7/katello-X.Y-rhel7/g
+:wqa
+
+git commit -a -m 'Updating rel-eng for X.Y'
+git push origin KATELLO-X.Y
+```
+
+Now we set master to start from the version after the one we are working on releasing X.Y+1, for each package, ensuring you are in the directory the spec file exists:
+
+```
+git checkout master
+```
+
+For any of the gems, update the gem version file and the spec file to X.Y.0 and make sure to reset the release to 1.
+
+```
+vim lib/katello/version.rb
+vim rubygem-katello.spec
+git commit -a -m 'Version bump to X.Y.0'
+```
```

Comment:
Or if this is just a example, we need something special here because it's incredibly and utterly confusing what's going on :confused: